packages/domain-picker/test/search-and-select-domain-suggestion.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { DataStatus, DomainSuggestion } from '@automattic/data-stores/src/domain-suggestions';
+
+/**
+ * Internal dependencies
+ */
+import '../matchMedia.mock';
+import DomainPicker from '../src/domain-picker';
+import { useDomainSuggestions } from '../src/hooks/use-domain-suggestions';
+
+// TODO: should we extract to separate mock data file/folder?
+const MOCK_SUGGESTION: DomainSuggestion = {
+	domain_name: 'example.com',
+	cost: '€15.00',
+	raw_price: 15,
+	currency_code: 'EUR',
+	relevance: 0.5,
+};
+
+const mockUseDomainSuggestionsResult: ReturnType< typeof useDomainSuggestions > = {
+	allDomainSuggestions: [ MOCK_SUGGESTION ],
+	errorMessage: null,
+	state: DataStatus.Success,
+	retryRequest: jest.fn(),
+};
+
+jest.mock( '../src/hooks/use-domain-suggestions', () => ( {
+	useDomainSuggestions: () => mockUseDomainSuggestionsResult,
+} ) );
+
+beforeEach( () => {
```

Comment:
Just for structure, I would opt to move this inside the `before`. WDYT?

Reply:
Assuming you meant `describe` instead of `before`, I did that in https://github.com/Automattic/wp-calypso/pull/50889/commits/155365f4c162572170429609a2dbcdff7b952289